pkg/process/checks/enabled_checks_test.go

```diff
@@ -90,11 +89,7 @@ func TestConnectionsCheck(t *testing.T) {
 		scfg.SetWithoutSource("system_probe_config.enabled", true)
 
 		enabledChecks := getEnabledChecks(t, cfg, scfg)
-		if runtime.GOOS == "darwin" {
-			assertNotContainsCheck(t, enabledChecks, ConnectionsCheckName)
-		} else {
-			assertContainsCheck(t, enabledChecks, ConnectionsCheckName)
-		}
+		assertContainsCheck(t, enabledChecks, ConnectionsCheckName)
```

Comment:
NPM is still not supported on macOS, so shouldn't this test not change?

Reply:
I think it's more a philosophical question, the `IsEnabled` method should return the status regarding the configuration according to its doc https://github.com/DataDog/datadog-agent/blob/5b9cfdee9baa5f8d1788d2e79d61432c10bc5c09/pkg/process/checks/checks.go#L47, so even if not supported it's actually enabled in this test.

By I agree with you that we could maybe do better. Do you think I should do a check in `adjust_{npm,usm,security}.go` files looking like:
```
if macos:
   cfg.set("product.enabled", false)
```
?

Reply:
What about having `checks.All` return different lists based on platform?

Reply:
That would be easier for me definitely, but I think this changes would require more input from the process team. Let me bring this to their slack to get a bit more feedback on this